compiler/rustc_ast/src/ast.rs

```diff
@@ -1695,7 +1695,7 @@ pub enum StrStyle {
 #[derive(Clone, Encodable, Decodable, Debug, HashStable_Generic)]
 pub struct Lit {
     /// The original literal token as written in source code.
-    pub token: token::Lit,
+    pub lit: token::Lit,
```

Comment:
Maybe I'm biased because I named it, but I don't think this needs changing.

`token::Lit` is a "token representation of a literal" or "literal token", so it's reasonable that it's named as token in AST representation of a literal.
The new name is more confusing - why is a literal inside a literal? How they are different? The answer is that one is an AST literal and another is a token literal.

Reply:
fwiw the subsequent `lit.lit`s driven by this rename made me do a double take

Reply:
The identifier `token` is used in *many* places to mean "a token", i.e. `Token`. I find it very misleading that it sometimes means "not a token". `token_lit` is a possible alternative.

Reply:
And as the commit message said: "This leads to some `lit.lit` chaining, but that's reasonable because we have an `ast::token::Lit` inside an `ast::Lit`". The variable/field names are reflecting the type names.

Reply:
I have pushed new code renaming it to `Lit::token_lit`. That avoids the use of `token` for a non-`Token`, which I don't like, and the use of `lit.lit`, which @calebcartwright doesn't like.

I think it's a good change, but if you're strongly opposed, @petrochenkov, I'll drop it, and then at least we can get the first two commits merged, hopefully.

Reply:
Okay, I'm still not very fond of the tautology, but it seems fine as a compromise.